test/topology_custom/test_raft_recovery_user_data.py

```diff
@@ -0,0 +1,153 @@
+    dead_servers = await manager.servers_add(3, config=cfg, property_file=property_file_dc2)
+    logging.info(f'Servers to syrvive majority loss: {live_servers}, servers to be killed: {dead_servers}')
+
+    cql, hosts = await manager.get_ready_cql(live_servers + dead_servers)
+    hosts = await wait_for_cql_and_get_hosts(cql, live_servers, time.time() + 60)
+
+    first_group0_id = (await cql.run_async(
+            "SELECT value FROM system.scylla_local WHERE key = 'raft_group0_id'"))[0].value
+
+    rf: int = 3
+    ks_name = unique_name()
+    restart_writes, stop_writes = await start_writes(cql, rf, ConsistencyLevel.LOCAL_QUORUM, 5, ks_name, True)
+
+    # Send some writes before we kill nodes.
+    logging.info('Sleeping for 1 s')
+    await asyncio.sleep(1)
+
+    logging.info(f'Killing {dead_servers}')
+    for srv in dead_servers:
+        await manager.server_stop(server_id=srv.server_id)
+
+    logging.info('Starting the recovery procedure')
+
+    logging.info(f'Restarting {live_servers}')
+    await manager.rolling_restart(live_servers)
+
+    logging.info(f'Deleting the persistent discovery state and group 0 ID on {live_servers}')
+    for h in hosts:
+        await delete_discovery_state_and_group0_id(cql, h)
+
+    recovery_leader_id = await manager.get_host_id(live_servers[0].server_id)
+    logging.info(f'Setting recovery leader to {live_servers[0].server_id} on {live_servers}')
+    for srv in live_servers:
+        await manager.server_update_config(srv.server_id, 'recovery_leader', recovery_leader_id)
+
+    logging.info(f'Restarting {live_servers}')
+    await manager.rolling_restart(live_servers)
+
+    # We stop writes and reconnect the driver because we perform a client request below
+    # (https://github.com/scylladb/python-driver/issues/295).
+    await stop_writes()
+
+    await reconnect_driver(manager)
+    cql, hosts = await manager.get_ready_cql(live_servers)
+
+    await restart_writes(cql)
+
+    if remove_dead_nodes_with == "remove":
+        # We must mark dead nodes as permanently dead so that they are ignored in topology commands. Without this step,
+        # ALTER KEYSPACE below would fail on the global token metadata barrier.
+        # For now, we do not have a specific API to mark nodes as dead so we use a work-around.
+        logging.info(f'Marking {dead_servers} as permanently dead')
+        await manager.remove_node(live_servers[0].server_id, dead_servers[0].server_id,
+                                  [dead_srv.ip_addr for dead_srv in dead_servers[1:]],
+                                  expected_error='Removenode failed')
+
+        logging.info(f'Desceasing RF of {ks_name} to 0 in dc2')
+        for i in range(1, rf + 1):
+            # ALTER KEYSPACE with tablets can decrease RF only by one.
+            await cql.run_async(f"""ALTER KEYSPACE {ks_name} WITH replication =
+                                    {{'class': 'NetworkTopologyStrategy', 'dc1': {rf}, 'dc2': {rf - i}}}""")
```

Comment:
It's quite bad UX that in case of DC loss we have to go through this weird procedure:
- first a failed `removenode` to mark nodes as dead
- then decreasing RF 1 by 1...
- and finally we can perform the actual `removenode`

cc @bhalevy @tgrabiec 

note: this applies also in scenarios where we don't need recovery from majority loss, e.g. when losing 1 out of 3 DCs.

Reply:
fortunately using replace is not that bad, and replace is usually preferable, so I guess it's one way to encourage the user to use replace instead of remove :)